notebooks/07_nsight_compute.ipynb

```diff
@@ -0,0 +1,252 @@
+    "--replay-mode application \\\n",
+    "bin/raja-perf.exe --variants [Base_CUDA OR Lambda_CUDA OR RAJA_CUDA] --tunings block_128 --size [1048576 OR 2097152] --repfact 0.01\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6ed93c84",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Map all files\n",
+    "ncu_dir = \"../data/ncu/\"\n",
+    "ncu_report_mapping = {}\n",
+    "variant = \"base_cuda\" # OR \"lambda_cuda\" OR \"raja_cuda\"\n",
+    "problem_sizes = [\"1M\", \"2M\"]\n",
+    "for problem_size in problem_sizes:\n",
+    "    full_path = f\"{ncu_dir}{variant}/{problem_size}/\"\n",
+    "    ncu_report_mapping[full_path+\"report.ncu-rep\"] = full_path+\"cuda_profile.cali\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "972c694b",
+   "metadata": {},
+   "source": [
+    "## 3. Read Calltree Profiles into Thicket\n",
+    "\n",
+    "The only performance metrics contained in the CUDA Activity Profile will be the CPU time `time` and the GPU time `time (gpu)`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "989da4d4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tk_cap = tt.Thicket.from_caliperreader(list(ncu_report_mapping.values()))\n",
+    "tk_cap.dataframe.head(20)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "94d9eff4",
+   "metadata": {},
+   "source": [
+    "## 4. Add NCU Data\n",
+    "\n",
+    "The Thicket `add_ncu` function takes one required argument and one optional arguement. The required argument `ncu_report_mapping` is the mapping from the NCU report file to the corresponding calltree profile run. The optional argument `chosen_metrics` allows for a subselection of the NCU performance metrics to add, since there can be hundreds of NCU performance metrics. By default we add all metrics."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "60ff359c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Add NCU to thicket\n",
+    "ncu_metrics = [\n",
+    "    \"gpu__time_duration.sum\",\n",
+    "    \"sm__throughput.avg.pct_of_peak_sustained_elapsed\",\n",
+    "    \"smsp__maximum_warps_avg_per_active_cycle\",\n",
+    "]\n",
+    "# Add in metrics\n",
+    "tk_cap.add_ncu(\n",
+    "    ncu_report_mapping=ncu_report_mapping, \n",
+    "    chosen_metrics=ncu_metrics,\n",
+    ")\n",
+    "tk_cap.dataframe.head(20)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "09e55c3a",
+   "metadata": {},
+   "source": [
+    "## 5. Add Problem Size to the Index\n",
+    "\n",
+    "We can add the problem size to the performance data index for clarity about which profile we are looking at."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b06b7e85",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tk_cap.metadata_column_to_perfdata(\"ProblemSizeRunParam\")\n",
```

Comment:
will this work with your new PR for adding multiple? let's make sure this won't break as soon as that feature goes in.

Reply:
The way the function is used stays the same, but the function name will change to `metadata_columns_to_perfdata`. With the plotting content added to the notebook I don't think this cell is necessary anyway, so I removed it. I had it in there because the notebook was short, but now I think we focus on the plotting.